test/github-issues/9229/issue-9229.ts

```diff
@@ -0,0 +1,88 @@
+import sinon from "sinon"
+import {
+    DataSourceOptions,
+    ConnectionOptionsReader,
+    DataSource,
+    DatabaseType,
+} from "../../../src"
+import {
+    setupTestingConnections,
+    createTestingConnections,
+    closeTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { CommandUtils } from "../../../src/commands/CommandUtils"
+import { MigrationGenerateCommand } from "../../../src/commands/MigrationGenerateCommand"
+import { TestEntity } from "./entity/test.entity"
+import { User } from "./entity/user"
+import path from "path"
+
+describe("github issues > #9229 for some reason, TypeORM generates a second migration file after running all migrations", () => {
+    let connections: DataSource[]
+    let connectionOptions: DataSourceOptions[]
+    let createFileStub: sinon.SinonStub
+    let getConnectionOptionsStub: sinon.SinonStub
+    let migrationGenerateCommand: MigrationGenerateCommand
+    let connectionOptionsReader: ConnectionOptionsReader
+    let baseConnectionOptions: DataSourceOptions
+
+    const enabledDrivers = ["postgres"] as DatabaseType[]
+
+    // simulate args: `npm run typeorm migration:run -- -n test-migration -d migration`
+    const testHandlerArgs = (options: Record<string, any>) => ({
```

Comment:
no need to use cli, you can run migrations through the api, e.g. dataSource.runMigrations

Reply:
This was actually doing migration generate, and I copy pasted that code from another test but I had forgot to change commentline. Now I changed query generation method and I don't use cli tool anymore.